### PR TITLE
FEAT: Load annulus from file, load reg files from IMPORT DATA

### LINKED ISSUE #2188: Import Data button to also load region files
**Reporter:** pllim

With image already loaded in Imviz, user click the "Import Data" button and select a region file (`*.reg`), the parser would be smart enough to run the `load_regions` method behind the scenes.

This would allow region file loading on standalone app with no API access. Therefore, this also allows non-interactive drawing of annulus in Imviz because annulus shape is now supported as of https://github.com/glue-viz/glue-astronomy/pull/90 .

It is already implemented in https://github.com/spacetelescope/jdaviz/pull/2164 but the relevant code needs to be plucked out for a clean PR. Also for annulus support only, also need new `glue-astronomy` release upstream (>0.7) and then update the pin in `pyproject.toml` (as needed).

[🐱](https://jira.stsci.edu/browse/JDAT-3376)

 --- 

 DISCLAIMER: This issue was autocreated by the Jdaviz Issue Creation Bot on behalf of the reporter. If any information is incorrect, please contact [Duy Nguyen](https://github.com/duytnguyendtn)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add two new features:

* fixes #2188
* load annulus region into a proper annulus ROI

### Blocked by

* glue-viz/glue#2403
* glue-viz/glue-astronomy#92

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ dependencies = [
     "specutils>=1.9",
     "specreduce>=1.3.0,<1.4.0",
     "photutils>=1.4",
-    "glue-astronomy>=0.7",
+    "glue-astronomy>=0.8",
```
Comment (pllim) [inline pyproject.toml:29]:
```suggestion
    "glue-astronomy>=0.9",
```

File: pyproject.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ dependencies = [
     "traitlets>=5.0.5",
     "bqplot>=0.12.37",
     "bqplot-image-gl>=1.4.11",
-    "glue-core>=1.6.0,!=1.9.0,!=1.10",
+    "glue-core>=1.10.1",
```
Comment (pllim) [inline pyproject.toml:15]:
```suggestion
    "glue-core>=1.11",
```

**Review by javerbukh — APPROVED** [review]
Looks good! Only question is do we want to keep using `get_interactive_regions()` over `get_subsets()`? I guess if we move away from using subsets in imviz the answer is "yes", but if we go another way it might be something to consider (out-of-scope for this PR though).

**Review by kecnry — APPROVED** [review]
Just one quick question, but otherwise the diff looks reasonable to me and the demo seems to work as expected!

If this doesn't currently catch the exception and show in the load file dialog, that would definitely be nice to include, if possible (like other errors where a wrong file extension is selected, etc).

File: CHANGES.rst [hunk 3]
```diff
@@ -12,6 +12,11 @@ Cubeviz
 Imviz
 ^^^^^
 
+- Added the ability to load DS9 region files (``.reg``) using the ``IMPORT DATA``
+  button. However, this only works after loading at least one image into Imviz. [#2201]
```
[the remark this answers is not in the document]
Comment (pllim) [reply CHANGES.rst:16]: The error would bubble up in the same was as any other error in Imviz parsing.

**Comment by pllim** [comment]
> get_interactive_regions() over get_subsets()

Good question and @kecnry had mentioned before in a few other occasions. I also agree it is out of scope here. Kyle, is there a ticket?

File: CHANGES.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (pllim) [reply CHANGES.rst:16]: Does this look good to you?

Screenshot 2023-06-07 161022

**Comment by pllim** [comment]
Since this has 2 approvals, I am going to merge. If you don't like how the loading error is presented, it also affects other parsing logic route and should be addressed separately.

Thanks for the reviews!

**Pull request merged**